### DISCUSSION TITLE
How to upgrade Node.js in Eclipse/.node directory

### DISCUSSION THREAD

**Discussion by wnm3** [mickaelistria/eclipse-bluesky#1529] [Q&A] [answered]
I installed a newer version of Node.js in the /Applications/Eclipse.app/Contents/Eclipse/.node directory and removed the old version, but running a js file as a Node Application causes the old version to be reinstalled.

I need to upgrade because our corporate security is bitching about the old version.

**Comment by vrubezhny** [comment, marked as answer]
If you're using WildWebDeveloper, it provides its own embedded version of Node.js (you can update from the latest snapshot using this p2 site: https://download.eclipse.org/wildwebdeveloper/snapshots). 

In case if for some reason you cannot update, you can instruct WWD to use a custom Node.js installation by setting up the `org.eclipse.wildwebdeveloper.nodeJSLocation` system property, see  https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/blob/master/TIPS_and_FAQ.md#use-embedded-nodejs for details.

**Comment by wnm3** [comment]
Thanks -- I tried adding this site and attempted installation but got this error:
An error occurred while collecting items to be installed
session context was:(profile=epp.package.jee, phase=org.eclipse.equinox.internal.p2.engine.phases.Collect, operand=, action=).
Problems downloading artifact: osgi.bundle,org.eclipse.wildwebdeveloper,1.1.5.202404111434.
SHA-512 hash is not as expected. Expected: 53f2484b7e1500f52237c8c0eb9350fb117cadb1f6b1a0c81667d9c93d60605a834a975f7aa5155e9dc05c1170fdd299bfd0b14ece428e65c83958117035168d and found db12367a6fdf586ba369295bf14530c70c955fe24e58bf269cffbae9fe5ef59fe68c908efe3f1285093546561943855a9ca648ede5ff922901e70539c2486610.

**Reply by vrubezhny** [reply]
It could be related to https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/issues/1505.

I can only suggest then to cancel the installation (hope it'll restore the previous version successfully) and try using the  `org.eclipse.wildwebdeveloper.nodeJSLocation` system property pointing to your custom Node.js installation folder.

**Comment by wnm3** [comment]
Thank you. I've tried that, but evidently, whichever plugin is updating the Eclipse/.node directory will replace the deleted version with the version it wants (which is outdated and causes vulnerability scanners to complain). In other words, the setting I added to eclipse.ini to define the desired location doesn't seem to have any effect.

**Reply by vrubezhny** [reply]
Can you move your Node.js installation (of the latest version you need to work with) somewhere out of `Eclipse/.node` directory?

**Comment by wnm3** [comment]
Yes, I've installed in /opt and set the definition to point to the node file like:
-Dorg.eclipse.wildwebdeveloper.nodeJSLocation=/opt/node-v21.7.3-linux-x64/bin/node
This worked on RHEL 8 

It also worked on my M1 MacBook using:
-Dorg.eclipse.wildwebdeveloper.nodeJSLocation=/Users/wnm3/.nvm/versions/node/v21.7.3/bin/node

Thank you.

**Comment by jmini** [comment]
Issue https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/issues/335 seems also relevant to this discussion.
